.changeset/brave-dots-breathe.md

```diff
@@ -0,0 +1,7 @@
+---
+"chainlink": minor
+---
+
+Added a new CLI command, `blocks find_lca,` which finds the latest block that is available in both the database and on the chain for the specified chain.
+Added a new CLI command, `node remove-blocks,` which removes all blocks and logs greater than or equal to the specified block number.
```

Comment:
What was the rationale behind placing one of these commands under `blocks` and the other one under `node`.  Seems like it might be more difficult to find them both if they're not in the same place?

Because these are both evm-specific commands, I think they should probably both go under `chainlink chains evm`... although I do notice there are some commands, such as `chainlink blocks replay` which are also evm-specific. 
(We should probably move that one under `chainlink chains evm` soon also. I'll mention that to those more involved with LOOP arch planning.)

Reply:
I agree that it's confusing. I've just tried to follow existing structure of the commands. 

Existing `blocks replay` interacts with the LogPoller and triggers it to replay blocks. `find-lca` - also interacts with the LogPoller and requests latest good block.  Thus it made sense to me to group them together.
At the same time `chains evm` simply lists available  evm chains.

Initially I've planed to have `remove-blocks` in the `blocks` section, but realised that it needs unique lock on the database.  `node`/`local` sections seems to contain all commands that must be executed on local node only. It also contains `rebroadcast-transactions`, which is EVM/TXM related but still resides here. I figured that all local commands must be grouped together to highlight that they are executed on the local node.

Foundations are the owners of this package, so It's up to you to decide where to put the commands.
Should I move them?

Reply:
Yeah, given that one of them is local and the other isn't I agree that it makes sense to have `remove-blocks` with `rebroadcast-transactions` and `find-lca` with `replay`. Unfortunately, I think we're going to want to move `remove-blocks` and `rebroadcast-transactions` soon to be under `chainlink blocks evm` and `chainlink node evm`.  So I'm not sure whether it's better to add them now in the same place and move them later, or just add them in the right place now and move the other two later. Asking rest of my team for some advice on that.

Reply:
@dhaidashenko Alright, yeah let's just add them here now and we can move them all together later.